src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final String SUPPLY_NAME = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyAmount = 0;
+        int buyAmount = 0;
+        String[] linePart;
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+            String value = reader.readLine();
+
+            while (value != null) {
+                linePart = value.split(",");
+                if (linePart[0].equals(SUPPLY_NAME)) {
+                    supplyAmount += Integer.parseInt(linePart[1]);
+                } else {
```

The code uses `else` to treat any non-`supply` operation as a `buy`. That silently accepts invalid operation names. The input format requires operation values to be "supply" or "buy" (see description). Validate the operation explicitly and throw a RuntimeException for unknown operation types instead of defaulting to `buy`.